consensus/hotstuff/signature/weighted_signature_aggregator.go

```diff
@@ -111,16 +112,13 @@ func (w *WeightedSignatureAggregator) hasSignature(signerID flow.Identifier) boo
 //  - engine.DuplicatedEntryError if the signer has been already added
 // The function is thread-safe.
 func (w *WeightedSignatureAggregator) TrustedAdd(signerID flow.Identifier, sig crypto.Signature) (uint64, error) {
-
-	currentWeight := w.TotalWeight()
-
 	info, found := w.idToInfo[signerID]
 	if !found {
-		return currentWeight, engine.NewInvalidInputErrorf("couldn't find signerID %s in the map", signerID)
+		return w.TotalWeight(), engine.NewInvalidInputErrorf("couldn't find signerID %s in the map", signerID)
```

Comment:
TotalWeight will acquire the lock, which we don't need unless the signer can't be found, which is an edge case

Reply:
https://github.com/onflow/flow-go/pull/1347/commits/31aafd421a2dc282bf339de657476c7732b497de